src/stores/pacbioPoolCreate.js

```diff
@@ -0,0 +1,954 @@
+    },
+
+    /**
+
+     * Asynchronously creates a pool with the given libraries and pools.
+     * Validates the libraries before proceeding.
+     * If the libraries are not valid, returns an error response.
+     * Otherwise, sends a request to create the pool and handles the response.
+     * Groups the included resources by type and extracts the barcode from the first tube.
+     *
+     * @async
+     * @returns {Promise<Object>} A promise that resolves to an object containing the success status, barcode, and any errors.
+     *
+     * @example
+     * // Create a pool
+     * const result = await createPool();
+     * console.log(result); // { success: true, barcode: 'barcode123', errors: [] }
+     */
+    async updatePool() {
+      const { libraries, pool } = this
+      if (!validate(libraries)) return { success: false, errors: 'The pool is invalid' }
+      const rootStore = useRootStore()
+      const request = rootStore.api.traction.pacbio.pools
+      const promise = request.update(payload({ libraries, pool }))
+      const { success, errors } = await handleResponse(promise)
+      return { success, errors }
+    },
+
+    /**
+     * Asynchronously populates libraries from a pool with the given ID.
+     * Sends a request to find the pool and includes all associated records.
+     * If the request is successful, populates the pool, libraries, requests, wells, plates, and tubes,
+     * selects the tag set, and selects all the tubes and plates.
+     *
+     * @async
+     * @param {number} poolId - The ID of the pool.
+     * @returns {Promise<Object>} A promise that resolves to an object containing the success status and any errors.
+     *
+     * @example
+     * // Populate libraries from a pool
+     * const result = await populateLibrariesFromPool(1);
+     * console.log(result); // { success: true, errors: [] }
+     */
+    async populateLibrariesFromPool(poolId) {
+      const rootStore = useRootStore()
+      const request = rootStore.api.traction.pacbio.pools
+      const promise = request.find({
+        id: poolId,
+        // We want to load *all* associated records, as otherwise we might be referencing them
+        // before they are loaded. Furthermore, if we start filtering the plates list at all,
+        // we may *never* load the relevant records.
+        // We load the other wells associated with the plate too, to ensure the remaining plate
+        // doesn't appear empty. This is especially important if the pool request finishes
+        // after the request for all plates, as otherwise the partial record will over-write
+        // the full one.
+        include:
+          'used_aliquots.tag.tag_set,used_aliquots.source_plate.wells.requests,used_aliquots.request.tube,tube',
+      })
+      const response = await handleResponse(promise)
+
+      const { success, data: { data, included = [] } = {}, errors = [] } = response
+
+      if (success) {
+        const {
+          library_pools,
```

Comment:
If used_aliquots are pulled back they will need to be added here as they will be an included group